File: src/Z.Core/System.String/_ExtractValueType/String.ExtractManyUInt16.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using System.Linq;
using System.Text.RegularExpressions;

public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that extracts all UInt16 from the string.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>All extracted UInt16.</returns>
    public static ushort[] ExtractManyUInt16(this string @this)
    {
        return Regex.Matches(@this, @"\d+")
            .Cast<Match>()
            .Select(x => Convert.ToUInt16(x.Value))
            .ToArray();
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/_ExtractValueType/String.ExtractManyUInt16.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Linq
Imports System.Text.RegularExpressions

Public Module Extensions_597

	''' <summary>
	'''     A string extension method that extracts all UInt16 from the string.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>All extracted UInt16.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ExtractManyUInt16(this As String) As UShort()
		Return Regex.Matches(this, "\d+").Cast(Of Match)().[Select](Function(x) Convert.ToUInt16(x.Value)).ToArray()
	End Function
End Module


